### PR TITLE
Paper revisions

### PR DESCRIPTION
**Pull request by yattias**
- Keep track of paper revisions
- Helps us figure out the number of citations changed since last time, open access status change, ...
- Can be used to display revisions of a paper on the client which is a future goal 

Closes https://github.com/ResearchHub/issues/issues/10

### CODE REVIEW

File: src/paper/migrations/0127_historicalpaper.py
Comment (yattias) [inline src/paper/migrations/0127_historicalpaper.py:1]: This migration is auto-generated by the simple history package

File: src/paper/related_models/paper_model.py [hunk 1]
```diff
@@ -59,6 +60,7 @@
 
 
 class Paper(AbstractGenericReactionModel):
+    history = HistoricalRecords()
```
Comment (yattias) [inline src/paper/related_models/paper_model.py:63]: This is how we tell the package to track history

**Review by koutst — APPROVED** [review]
LGTM

File: src/pyproject.toml [hunk 2]
```diff
@@ -83,6 +83,7 @@ urllib3 = "1.26.18"
 aiohttp = "3.9.5"
 cryptography = "42.0.5"
 requests-mock = "^1.12.1"
+django-simple-history = "^3.7.0"
```
Comment (gzurowski) [inline src/pyproject.toml:86]: I suggest to target a specific version instead of a range.

Comment (yattias) [reply src/pyproject.toml]: Good catch, I thought I did

**Pull request merged**